Apps/Playground/Android/app/src/main/cpp/BabylonNativeNetwork.cpp

```diff
@@ -0,0 +1,140 @@
+{
+    JNIEnv* AcquireEnvFromJVM(JavaVM* jvm)
+    {
+        JNIEnv * env;
+        int getEnvStat = jvm->GetEnv((void **)&env, JNI_VERSION_1_6);
+        if (getEnvStat == JNI_EDETACHED)
+        {
+            log("GetEnv: not attached");
+            if (g_jvm->AttachCurrentThread(&env, NULL) != 0)
+            {
+                log("Failed to attach Env");
+                return nullptr;
+            }
+        } else if (getEnvStat == JNI_EVERSION)
+        {
+            log("GetEnv: version not supported");
+            return nullptr;
+        }
+        return env;
+    }
+
+    template<typename T> bool AndroidLoadAsset(const std::string& url, T& data)
+    {
+        AAsset* asset = AAssetManager_open(g_assetMgrNative, url.c_str(),
+            AASSET_MODE_UNKNOWN);
+        if (asset)
+        {
+            size_t size = AAsset_getLength64(asset);
+            data.resize(size);
+            AAsset_read(asset, data.data(), size);
+            AAsset_close(asset);
+            return true;
+        }
+        return false;
+    }
+}
+
+namespace Babylon
+{
+    std::string AndroidGetAbsoluteUrl(const std::string& url, const std::string& rootUrl)
+    {
+        AAsset* asset = AAssetManager_open(g_assetMgrNative, url.c_str(),
+            AASSET_MODE_UNKNOWN);
+        if (asset)
+        {
+            AAsset_close(asset);
+            return url;
+        }
+        if (!rootUrl.empty())
+        {
+            return rootUrl + "/" + url;
+        }
+        return url;
+    }
+
+    void AndroidLoadURLFunction(const std::string& url, std::function<void(const std::string& content)> contentHandler)
+    {
+        std::string content;
+        if (AndroidLoadAsset(url, content))
+        {
+            contentHandler(content);
+        }
+        else
+        {
+            DispatchLooperFunction<DispatchPayloadDirect>(g_mainThreadLooper, [url, contentHandler = std::move(contentHandler)](){
+                JNIEnv * env = AcquireEnvFromJVM(g_jvm);
+                assert(env);
+                auto urlString = env->NewStringUTF(url.c_str());
+                jstring javaString = (jstring)env->CallStaticObjectMethod(networkClass, loadURLStringMethod, urlString);
+                const char *nativeString = env->GetStringUTFChars(javaString, 0);
+                std::string content(nativeString);
+                contentHandler(content);
+                env->ReleaseStringUTFChars(javaString, nativeString);
+            });
+        }
+    }
+
+    void AndroidLoadURLFunction(const std::string& url, std::function<void(const std::vector<uint8_t>& content)> contentHandler)
+    {
+        std::vector<uint8_t> content;
+        if (AndroidLoadAsset(url, content))
+        {
+            contentHandler(content);
+        }
+        else
+        {
+            DispatchLooperFunction<DispatchPayloadDirect>(g_mainThreadLooper, [url, contentHandler = std::move(contentHandler)](){
+                JNIEnv * env = AcquireEnvFromJVM(g_jvm);
```

Comment:
Which thread would this be getting called from? Is it the JS thread? If so, this part of the change will overlap with a more generalized solution that I believe @bghgary was going to implement for getting the JVM attached to the JS thread as needed (we need to do this for the ARCore integration).

Reply:
That's right. @CedricGuillemet put a comment on DipatchLooperFunction letting me know the code to put in arcana.

Reply:
I wasn't talking about the looper scheduler, I was talking about attaching the JVM to the JS thread.

Reply:
Oh, right. Yeah, we should use the new code that in the arcore branch.